rpc/src/server/api.rs

```diff
@@ -55,6 +69,13 @@ impl api_server::Api for RpcApi {
     ) -> Result<Response<SyncStateResponse>, Status> {
         self.store.clone().sync_state(request).await
     }
+
+    async fn publish_proven_transaction(
+        &self,
+        request: Request<PublishProvenTransactionRequest>,
+    ) -> Result<tonic::Response<PublishProvenTransactionResponse>, Status> {
+        self.block_producer.clone().publish_proven_transaction(request).await
```

Comment:
Is `clone()` what we want to do here, or rather should we be putting this behind an `Arc<Mutex<_>>`? I couldn't find it in the tonic docs

Reply:
Yes. Ref: https://docs.rs/tonic/latest/tonic/client/index.html#concurrent-usage